File: src/main/java/com/leetcode/easy/A203_Remove_Linked_List_Elements.java
package com.leetcode.easy;

import com.publicClass.ListNode;

// 203. 移除链表元素
public class A203_Remove_Linked_List_Elements {
    public ListNode removeElements(ListNode head, int val) {
        ListNode front = new ListNode(0);
        front.next = head;
        ListNode pre = front;
        while (pre.next != null) {
            if (pre.next.val == val) {
                pre.next = pre.next.next;
            } else
                pre = pre.next;
        }
        return front.next;
    }

}

File: src/main/java/com/leetcode/easy/_0203_Remove_Linked_List_Elements/A203_Remove_Linked_List_Elements.cpp
#include <bits/stdc++.h>
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
// 递归法
class Solution {
public:
    ListNode* removeElements(ListNode* &head, int val) {
        ListNode* tmp;
        if (head == NULL)
            return head;
        if (head->val == val) {
            tmp = head;
            head = head->next;
            delete tmp;
            removeElements(head, val);
        } else {
            removeElements(head->next, val);
        }
        return head;
    }
};